packages/sdk/src/modules/FundOperations.ts

```diff
@@ -28,6 +28,18 @@ export function withFundOperations<TBase extends MidasBaseConstructor>(Base: TBa
       return { gasWEI, gasPrice, estimatedGas };
     }
 
+    async approve(cTokenAddress: string, underlyingTokenAddress: string) {
+      const token = getContract(underlyingTokenAddress, this.artifacts.EIP20Interface.abi, this.signer);
+      const max = BigNumber.from(2).pow(BigNumber.from(256)).sub(constants.One);
+      const tx = await token.approve(cTokenAddress, max);
```

Comment:
```suggestion
return token.approve(cTokenAddress, max);
```

Lets just return the actual transaction. This way the user of this function can decide what to do with it and how to handle the error.

In 39 the code just sends back a generic error code obfuscating the actual error code of the failed transaction.

Reply:
Returns tx, resolved